app/models/solidus_stripe/payment_method.rb

```diff
@@ -8,10 +8,28 @@ class PaymentMethod < ::Spree::PaymentMethod
     preference :stripe_intents_flow, :string, default: 'setup'
     preference :skip_confirmation_for_payment_intent, :boolean, default: true
 
+    # @attribute [rw] preferred_webhook_endpoint_signing_secret The webhook endpoint signing secret
+    #  for this payment method.
+    # @see https://stripe.com/docs/webhooks/signatures
+    preference :webhook_endpoint_signing_secret, :string
+
     validates :available_to_admin, inclusion: { in: [false] }
     validates :preferred_setup_future_usage, inclusion: { in: ['', 'on_session', 'off_session'] }
     validates :preferred_stripe_intents_flow, inclusion: { in: ['payment', 'setup'] }
 
+    has_one :webhook_endpoint,
+      class_name: 'SolidusStripe::WebhookEndpoint',
+      inverse_of: :payment_method,
+      dependent: :destroy
```

Comment:
Oh, I think it's for finding it quickly from the DB instead of loading up all the payment methods just to get the preference value, because it's stored in a YAML, is that correct?

If that's the case I think it would be ok to expect a store only to have a handful of payment methods and shouldn't have any impact on performance. Thoughts?

Reply:
@elia, thanks for that suggestion. Yeah, I considered having it as a preference, but besides what you pointed out, there's no way to ensure uniqueness, and it requires user action for something that should be automatic. What do you think?

Reply:
> can be simply ensured through a validation

Is that true? I mean, you can create a valid payment method record and later change the preferences via env variables. I think Solidus wouldn't validate that on app initialization, but as you say, it'd be clumsy and cumbersome.

Reply:
In the end this is probably more of an upstream problem, in which prefs are a weird and not fully consistent tool. I took note and you're right, this is a good solution given the context 👍